shared/lib/ArappFile.js

```diff
@@ -0,0 +1,44 @@
+const fs = require('fs')
+const path = require('path')
+
+const FILE_NAME = 'arapp.json'
+const LOCAL_FILE_NAME = 'arapp_local.json'
+const DEFAULT_ARAPP_FILE = { environments: {} }
+
+module.exports = web3 => {
+  const { isLocalNetwork, getNetworkName } = require('./Network')(web3)
+  
+  const ArappFile = {
+    async fileName() {
+      return (await isLocalNetwork()) ? LOCAL_FILE_NAME : FILE_NAME
+    },
+
+    async filePath() {
+      return path.resolve(await ArappFile.fileName())
+    },
+
+    async file() {
+      const filePath = await ArappFile.filePath()
+      const file = fs.existsSync(filePath) ? require(filePath) : DEFAULT_ARAPP_FILE
+      if (!file.environments) file.environments = {}
+      return file
+    },
+
+    async deployedAddresses() {
+      const network = await getNetworkName()
+      const file = await ArappFile.file()
+      return file.environments[network] || {}
+    },
+
+    async write(appName, address, contractName, registry) {
+      const network = await getNetworkName()
+      const data = await ArappFile.file()
+      data.path = `contracts/${contractName}.sol`
+      if (data.environments === undefined) data.environments = {}
+      data.environments[network] = { appName, address, network, registry }
+      fs.writeFileSync(await ArappFile.filePath(), JSON.stringify(data, null, 2))
+    }
+  }
+
+  return ArappFile
```

Comment:
In these modules, I find the extra level of scoping a bit redundant, since we never need to access `this`.

Another problem that _could_ happen is someone accidentally changing a declaration at runtime (e.g. requiring `ArappFile` and then modifying `ArappFile.file` by accident).

What do you think about declaring them separately and only exporting them at the end?

```js
module.exports = web3 => {
  const { isLocalNetwork, getNetworkName } = require('./Network')(web3)

  async fileName() {
    return (await isLocalNetwork()) ? LOCAL_FILE_NAME : FILE_NAME
  }

  async filePath() {
    return path.resolve(await fileName())
  }

  ...

  return {
    fileName,
    filePath,
  }
}
```